src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -973,7 +976,6 @@ internal async Task Cleanup(bool async, bool connectionClosing = false, bool isD
 
             State = ReaderState.Closed;
             Command.State = CommandState.Idle;
-            Connector.CurrentReader = null;
```

Comment:
We can still leave this here (instead of putting it in NpgsqlCommand.ExecuteReader), no?